pkg/kubectl/cmd/apply_test.go

```diff
@@ -279,6 +296,85 @@ func TestApplyObject(t *testing.T) {
 	}
 }
 
+// TestApplyTPR uses apply commad with ThirdPartyResources.
+func TestApplyTPR(t *testing.T) {
+	initTestErrorHandler(t)
+	nameTPR := "test-tpr.unit-test.test.com"
+	pathTPR := "/thirdpartyresources/" + nameTPR
+	currentTPR := readTPRFromFile(t, filenameTPR)
+
+	verifiedPatch := false
+
+	f, tf, _, ns := cmdtesting.NewAPIFactory()
+	tf.Printer = &testPrinter{}
+	tf.JSONEncoder = testapi.Extensions.Codec()
+	tf.UnstructuredClient = &fake.RESTClient{
+		ContentConfig: restclient.ContentConfig{
+			ContentType: runtime.ContentTypeJSON,
+			GroupVersion: &schema.GroupVersion{
+				Group:   extensions.GroupName,
+				Version: "v1beta1",
+			},
+			NegotiatedSerializer: ns,
+		},
+		NegotiatedSerializer: ns,
+		Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
+			switch p, m := req.URL.Path, req.Method; {
+			case p == pathTPR && m == "GET":
+				bodyTPR := ioutil.NopCloser(bytes.NewReader(currentTPR))
```

Comment:
I see. Good catch thanks for pointing that out. 

The annotateRuntimeObject function was already doing that. However in this PR I have implemented my own readTPRFromFile function. The annotateRuntimeObject was using the testapi.Default.Codec(). I need testapi.extensions.Codec(). 

I will either use a similar approach to the annotateRuntimeObject function or try to re-use the annotateRuntimeObject funciton.